platform/keysystem/keyserver/keyapi/keyserver.go

```diff
@@ -45,12 +56,45 @@ func apiToHTTP(ks Keyserver, logger *log.Logger) http.Handler {
 	return mux
 }
 
+func generateServerCertificate(ctx *config.Context) (tls.Certificate, error) {
+	// TODO: refactor out the key generation pattern
+	privateKey, err := rsa.GenerateKey(rand.Reader, TemporaryCertificateBits)
+	if err != nil {
+		return tls.Certificate{}, err
+	}
+	contents := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
+	})
+	csr, err := csrutil.BuildTLSCSR(contents)
+	if err != nil {
+		return tls.Certificate{}, errors.Wrap(err, "while generating CSR")
+	}
+	// TODO: figure out the right period of time for this certificate to be valid
+	cert, err := ctx.ClusterTLS.Sign(string(csr), true, time.Hour*24, "keyserver-autogen-tls", []string{ctx.KeyserverDNS})
```

Comment:
the commit message says this will be generated every time the keyserver starts up.  When does it regenerate it?

Reply:
It doesn't currently get regenerated except on startup. I should probably fix that; not renewing it is an oversight.